daemon/events.go

```diff
@@ -130,3 +141,181 @@ func copyAttributes(attributes, labels map[string]string) {
 		attributes[k] = v
 	}
 }
+
+// ProcessClusterNotifications gets changes from store and add them to event list
+func (daemon *Daemon) ProcessClusterNotifications(eventCh chan *swarmapi.WatchMessage) {
+	for {
+		select {
+		case message, ok := <-eventCh:
+			if !ok {
+				return
+			}
+			daemon.generateClusterEvent(message)
```

Comment:
Is there any risk of a panic or deadlock if `generateClusterEvent` causes `daemon.EventsService.PublishMessage` to be called during the daemon's shutdown process? This is a separate goroutine that might continue receiving events as the daemon shuts down.